File: frontend/src/api/compliance.ts
import { api } from './client';

export interface ComplianceItem {
  id: string;
  workspace_id: string;
  user_id: string;
  item_type: string;
  title: string;
  reference_number: string | null;
  issue_date: string | null;
  expiry_date: string | null;
  status: string;
  alert_days: number[];
  document_id: string | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export const complianceApi = {
  list: (workspaceId: string, params?: { user_id?: string; status?: string; item_type?: string }) =>
    api.get<ComplianceItem[]>(`/workspaces/${workspaceId}/compliance`, { params }),

  create: (workspaceId: string, data: Partial<ComplianceItem>) =>
    api.post<ComplianceItem>(`/workspaces/${workspaceId}/compliance`, data),

  update: (workspaceId: string, itemId: string, data: Partial<ComplianceItem>) =>
    api.put<ComplianceItem>(`/workspaces/${workspaceId}/compliance/${itemId}`, data),

  delete: (workspaceId: string, itemId: string) =>
    api.delete(`/workspaces/${workspaceId}/compliance/${itemId}`),
};

File: backend/app/schemas/compliance.py
import uuid
from datetime import date, datetime

from pydantic import BaseModel


class ComplianceItemCreate(BaseModel):
    user_id: uuid.UUID
    item_type: str
    title: str
    reference_number: str | None = None
    issue_date: date | None = None
    expiry_date: date | None = None
    alert_days: list[int] | None = None
    document_id: uuid.UUID | None = None
    notes: str | None = None


class ComplianceItemUpdate(BaseModel):
    item_type: str | None = None
    title: str | None = None
    reference_number: str | None = None
    issue_date: date | None = None
    expiry_date: date | None = None
    status: str | None = None
    alert_days: list[int] | None = None
    document_id: uuid.UUID | None = None
    notes: str | None = None


class ComplianceItemResponse(BaseModel):
    id: uuid.UUID
    workspace_id: uuid.UUID
    user_id: uuid.UUID
    item_type: str
    title: str
    reference_number: str | None = None
    issue_date: date | None = None
    expiry_date: date | None = None
    status: str
    alert_days: list[int] | None = None
    document_id: uuid.UUID | None = None
    notes: str | None = None
    created_at: datetime
    updated_at: datetime
    model_config = {"from_attributes": True}
